RDK-57972 AAMP Federated Release for Sprint 2506

paulpandian25 opened the pull request:
Feature/rdk 57972 federated 2506 develop

From dev_sprint_25_2 : 92dd326d5387861b4ead4a288aaea1d62838f2ef revision.

Deepikasri1234 committed:
RDK-56217 Decouple DRM License dependencies in AAMP

Reason for change: Decouple License dependencies to seperate folder.
Test Procedure: Refer Ticket
Risks: Low
Priority: P1

Change-Id: I38c93ca222ccd2dfc594ce66858f07e659d2dc4b
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Commit:
RDKAAMP-4027:  HLS SLE Thumbnail support

Reason for change : added support for Thumbnails for SLE and Linear content for seek opeartions
Risks: LOW
Test Procedure: updated in ticket
Priority: P1

Change-Id: Icc2917aff025da629d280ebd4b852b33e215acf8
Signed-off-by: Gnanesha <Gnanesha_Thimmegowda@comcast.com>

Commit:
RDKAAMP-4093 : Black screen when playing VoD with CDAI

Reason for change: IARM/RFC migration cdai crash fix
Test Procedure: Mentioned in ticket
Risks: None
Signed-off-by: naren_ramesh@cable.comcast.com

Change-Id: If61a9734342415c28c3ecbf50fec94cf26f7721a

Commit:
RDKAAMP-4128 :  [HLS] Support Header/Body response event for DRM failures as object within header response event

Reason for change : send drm meta data upon license request failure
Risks: Low
Test Procedure: updated in ticket
Priority: P1

Change-Id: I6dc96e33fa4198063b79f1aa3e032adf384a9c53
Signed-off-by: Gnanesha <Gnanesha_Thimmegowda@comcast.com>

nudiya450 committed:
VPLAY-9208: [L1] New test case for SelectSourceOrAdPeriod

Reason for change: Test case for SelectSourceOrAdPeriod where the ad in the landing period is not yet resolved,
but player is in IN_ADBREAK_WAIT2CATCHUP
Risks: Low
Test Procedure: Test with
Priority: P1

Change-Id: I5130e5ae15cf09b7d0f069a9f3276539a5dd61c0
Signed-off-by: Nandakishor Udiyannur<Nandakishor_udiyannurmana@comcast.com>

VPLAY-8860: ads played out on the 1st play but same spot got substituted by DAI on rewind

Reason for Change: CDAI ad is not being delivered soon enough to be
used on 1st play, but it does arrive and is used on 2nd play. This
change to invalidate the AD if it is not used on 1st play hence
it will not be used in subsequent play.

Test Procedure. See jira

Risk:Medium

Change-Id: Ia627d818885f9de2d9e3236e3fa685c1d022c62e
Signed-off-by: Nandakishor Udiyannur<Nandakishor_udiyannurmana@comcast.com>

hthakkar-synamedia committed:
VPLAY-8905  AAMP TSB - Trickmodes ff to live SLD

Reason for change: - For SLD, while running L2 tests, the
player freezes while tranistioning from ff to live. This is because
seek position use current time clock, while the L2 test streams
are harvested sometime back, leading to freeze.

The solution is to use steady clock instead of current time clock.

This commit also fixes a bug, wherein during forward trick play,
tsb content available at the time of trick play was being considered.
Content downloaded after initiating trick play was not taken into account.

The solution is to calculate a delta at the tune based on steady clock,
live offset and latest time period. When the trick play is requested,
the position at which EOS is requested is based on delta calcuated at
tune, live offset and trick mode speed.

Test Procedure: Mentioned in the ticket

Risks: - Medium

Signed-off by: - Himanshu Thakkar<hthakkar@synamedia.com>

Change-Id: I35208428aa6a4d8d99f28a28124d82724c397ed1

aahmed878 committed:
RDK-56189: Aamp as a Plugin Deprecation[DeadCode Removal]

Reason for change: Removing uncompiled dead code as part of RDK-56189

Test Procedure: N/A
Risks: Low
Priority: P1

Change-Id: I9c681be30018289f545a7db8e3d07c60d6e3f2a7
Signed-off-by: Alsameema.A <Alsameema_AhmedAnsar@Comcast.Com>

Commit:
VPLAY-9237: [AAMP] StreamAbstractionAAMP::GetVideoBitrate Crash

Reason for change: Null check before accessing mpStreamAbstractionAAMP
instance
Test Procedure: As mentioned in the ticket
Risks: None

Change-Id: Ia833004a3da5170592074289767b7ed10792bb37
Signed-off-by: Vinish K B <Vinish_KorasseryBalakrishnan@comcast.com>

pstroffolino committed:
VPLAY-9243 override mdhd.duration and deprecate segment.stop override

Reason for change: tech debt - more elegant way to avoid unwanted qtdemux stop
Test Procedure: do no harm testing; refer ticket for original issue
Risks: Low

Change-Id: Id073bc911838f6e12523763a950215a2d4096401

Reshma-001 committed:
VPLAY-9197: L1 for onAdEvent function covering trickplay cases

Reason for change: Adding L1 test cases for
onAdEvent for covering rewind case.
Test Procedure: Run L1 tests for onAdEvent
Risks: Low

Change-Id: I6bd9cace4a462b967521a393abe4c7715c9d6990
Signed-off-by: Reshma Raphael<Reshma_Raphael@comcast.com>

kd-syna committed:
VPLAY-9092: AAMP UVE API Update Request: "enablePTSReStamp" only valid for DASH (not HLS)

Reason for change: API documentation unclear about restrictions for the "enablePTSReStamp" configuration option.

Test Procedure: Observe the restriction in the documentation.

Risks: None
Change-Id: I245e793f6b9bf896b39b3846c111fa53b7b0c6e8

pstroffolino committed:
VPLAY-9237 spelling fix: trickplays -> trickplay

Reason for change: godspell-flagged spelling
Test Procedure: no godspell warning
Risks: None

Change-Id: Ie7653c88c87e8a03d788ddbac7e9d944dca85e49

Commit:
VPLAY-8908  AAMP TSB - LLD - TSB writing stops when paused on local TSB

Reason for change: With AAMP TSB active then during pause the segment
fetch must continue to write to TSB. Segment inject from TSB reader
should stop once enough-data signal is received.

Test procedure: Existing L1 and L2 tests. Could not add a specific
L2 test as enough-data is not generated in L2. Test at full
stack using description in jira.

Risks: Medium

Change-Id: I6a6cc5afdb938797f78a350d443073a0a7f6d9f8

psiva01 committed:
VPLAY-9254: Observed app crash with fingerprint 3540680

Reason for change: Reworked NotifyEOSReached() to avoid crash
when mpStreamAbstractionAAMP gets deleted by SetRate b/w checking for
mpStreamAbstractionAAMP not null & IsEOSReached() in different lines.
Test Procedure: Refer Jira
Risks: Low

Change-Id: I121c5e5d528c58f4626ceebcce35544a871a4af6
Signed-off-by: Sivasubramanian <sivasubramanian.patchaiperumal@ltts.com>

jfagunde committed:
VPLAY-9236 Set rate in L1 test PrivAampTests.stopTest

Reason for change: L1 test PrivAampTests.stopTest fails intermittently

Changes:
* Set rate to PLAY so stopTest verifies pausing the playback
* Add small delays to allow thread switching

Test Procedure: run L1 PrivAampTests

Risks: Low

Change-Id: I6d6f4fa58ad6767336ee0d03d3aebd33e166ba4c
Signed-off-by: Jose Fagundez <jfagunde@synamedia.com>

jfagunde committed:
VPLAY-9267 Add timestamp and thread ID to L1 log lines

Reason for change: Help investigate L1 test failures

Changes:
* Add timestamp to L1 test log lines
* Do not rely on AampUtils  to print thread ID in L1 test log
* Delete commented out code
* Delete ENABLE_TEST_LOGGING define, since it is always on for L1 tests

Test Procedure: run L1 test with --verbose option

Risks: Low

Change-Id: I04bacb139ed3e8dff36869259284be6b09e0334b
Signed-off-by: Jose Fagundez <jfagunde@synamedia.com>

pstroffolino committed:
VPLAY-9265 simplify "allowed trickplay speed" handling

Reason for change: deprecate IsValidRate, common getSupportedRates
Test Procedure: l1 tests and no x1 bugs related to trickplay
Risks: Low

Change-Id: I0f76d768e0c21e498bc06b86aacadda7a793b9a2

pstroffolino committed:
VPLAY-9100 logging macro redefinition in middleware layer

Reason for change: avoid conflicting definitions in middleware header and SubtecChannel.cpp
Test Procedure: build in xcode
Risks: Low

Change-Id: Icf699aa2e80fba40b46b59881db1219b1813d3b4

pstroffolino committed:
VPLAY-9268 compilation warnings surfaced by latest xcode

Reason for change: avoid  xcode warnings in this area
Test Procedure: warnings no longer seen in xcode
Risks: Low - no functional change

Change-Id: Ia21937640e8a1c3e331ebc6186d5b0af499ff878

DomSyna committed:
VPLAY-9327 WPEWebProcess crash with fingerprint 76741189

Reason for change:
WebProcess crashes in AampCache::Remove due to assertion `iter2 != cache.end()'

This is because 2 different URLs resolve to the same effective URL. When
the first is removed from the cache the associated effective URL is also
removed. As a result when the second URL is removed the associated
effective URL no longer exists.

This change checks if any other URL reference the effetive URL before
removing it.

Test Procedure: Run L1 tests, and confirm at full stack the issue is
resolved.

Risks: Low

Signed-off-by: Dominic Holden <doholden@synamedia.com>

Change-Id: Ia63449cf82ceb22e6223643b3d6b02841bdeb977

jfagunde committed:
VPLAY-9279 Modify MockAampJsonObject to match MockPlayerJsonObject

Reason for change: Make it easier to port changes to branches like 8.1

Changes:
* Rename MockAampJsonObject get_string() to get() to match
  MockPlayerJsonObject

Test Procedure: Run L1 tests

Risks: Low

Change-Id: I54bf44f3f68acec52dca1e4994bb021927e7cacd
Signed-off-by: Jose Fagundez <jfagunde@synamedia.com>

molakalapalliharipriya committed:
VPLAY-9316 : [CDVR] No progress event and seek failures after Hot To Cold CDVR transition

Reason for change: positions values are not calculated when manifest transitions from dynamic to static

Test Procedure: Test case updated in the ticket
Risks: Medium

Change-Id: I1f05691195186fcabd1e159f3aeb2bdaab9209f8
Signed-off-by: haripriya_molakalapalli <haripriya_molakalapalli@comcast.com>

pstroffolino committed:
VPLAY-9303 parse error if HLS playlist ends with URI=".."

Reason for change: fix parsing edge case
Test Procedure: refer ticket
Risks: Low

Change-Id: Iaf079651e78e231dbd35c02dd14b2cbeb5e8221b

Commit:
VPLAY-9307 AAMP TSB crash in L3 2016

Reason for change: Race condition when starting
injector thread causes crash. Add thread safe
mutex around start/stop of injector thread

Test procedure: L3 test 2016 and confirm no
assert is occuring.

Risk: Low

Change-Id: I970f4be491ba911b61abbcb17ad3d6d60bb9434f

anshephe committed:
VPLAY-9314  AAMP TSB - TSB writing stops following an advert if paused on live

Reason for change:

Don't wait for discontinuity in FetcherLoop if AAMP TSB enabled and paused

Test Procedure:
Tested at full stack
Run AAMP L1 and L2 tests

Risks: Low

Signed-off-by: Andrew Shepherd<anshephe@synamedia.com>

Change-Id: I5529d5ac4907083bb536fa1bc39edd9b446b7fa9

psiva01 committed:
VPLAY-9239: Video freezes when performing seek operations

Reason for change: Modify NotifyFragmentCachingComplete() to set buffering_target_state as well to PLAYING if pendingPlayState is set. This will avoid the timing issue when SetRate(1) moves the pipeline to PLAYING between SeekWhilePaused & its buffering timeout callback. So, when buffering timeout callback comes, it sets the player to PAUSED (Previous state set by SeekWhilePaused) instead PLAYING.
Test Procedure: Refer Jira
Risks: Medium

Change-Id: I7cda20d8bf9a0e892d89a1f3f8d0f4936c49e0e5
Signed-off-by: Sivasubramanian <sivasubramanian.patchaiperumal@ltts.com>

Commit:
VPLAY-9326  OSX AAMP simulator build failure after OSX updates

Reason for change: To update the correct version of openssl to install in the install-aamp script
Risks: Low
Priority: P0

DEPS_BRANCH=rdk/components/generic/gst-plugins-rdk-aamp:dev_sprint_25_1
DEPS_BRANCH=rdk/components/generic/aampabr:dev_sprint_25_1
DEPS_BRANCH=rdk/components/generic/aampmetrics:dev_sprint_25_1

Change-Id: I6dc8b1e82a8d66853eef3397762d9c3ce79c6820
Signed-off-by: Rajat <Rajat_Yadav2@comcast.com>

Commit:
Reapply "VPLAY-9571: Progress reports broken when PTS restamping enabled"

This reverts commit 89e7fe2ee16b5a5b078a4af74953346ff27d5c3d.

Change-Id: I5adeb5fba29fa47fb25a1109acfc79fa8b8aa6bf

pstroffolino committed:
Revert "RDKAAMP-4128 :  [HLS] Support Header/Body response event for DRM failures as object within header response event"

This reverts commit 69e6897d7db8a460534a66e515e798eb29fd3b91.

Reason for revert: causing regression VPLAY-9353

Change-Id: I7f41f480bb39e6fa401842d3c5130a904bce064a

Commit:
VPLAY-9337: [HLS] Support Header/Body response event for DRM failures as object within header response event

Reason for change : send drm meta data upon license request failure
w/o any crash regression. Added check for invalid response which is
not NULL & caused crash in VPLAY-9352 & VPLAY-9353.
Test Procedure: Refer ticket
Risks: Medium

Revert "Revert "RDKAAMP-4128 :  [HLS] Support Header/Body response event for DRM failures as object within header response event""
This reverts commit a5c276d0f2a91e2a411d247d69c2aa3bef23aa36.

Change-Id: I4a0a7c01007194b34b93ddf07b0d4a99f8c44440
Signed-off-by: Sivasubramanian <sivasubramanian.patchaiperumal@ltts.com>

VPLAY-9354  Address L1/L2 Test Gap for escaped sprint regression VPLAY-9353

Reason for change: l1 test for aamp_CreateBodyResponseJSObject
Test Procedure: needs to be delivered along with fix for regression
Risks: Low

Change-Id: I7878549a19af9df4dba5829e83f92c61f82d8697

Change-Id: I4a0a7c01007194b34b93ddf07b0d4a99f8c44440
Signed-off-by: Sivasubramanian <sivasubramanian.patchaiperumal@ltts.com>

kd-syna committed:
VPLAY-8956: Print PlayerId in TSB log messages

Reason for change: Threads created by the TSB library did not have
the thread-local gPlayerId field set and were therefore reporting a
PlayerId of -1 in log messages. This change propogates the PlayerId
of the instance which creates a TSB::Store to threads created by
that Store.

Test Procedure: L1 + L2 tests to confirm no regressions. Check logs for TSB output (specifically, from "Flusher") to ensure the PlayerId is not -1.

Risks: Very Low/None. Affects log output only.
Change-Id: I787f16836b898f38e5f941226d8f0c4ebf7f5c55

psiva01 committed:
VPLAY-9254: Observed app crash with fingerprint 3540680

Reason for change: Fixed L3 regression due to deadlock
Test Procedure: Refer Jira
Risks: Low

Change-Id: I46c6d6e1ee3f9b3cff8f297a6c90305ac3b4a20e
Signed-off-by: Sivasubramanian <sivasubramanian.patchaiperumal@ltts.com>

Commit:
VPLAY-9389: Update AAMP version to 7.05

Reason for change: Update AAMP version to 7.05
Test Procedure: Validate version in IP_AAMP_TUNETIME
Risks: None

Change-Id: I4a92f9304a1ff207df1cc4589de5aeff93f8b239
Signed-off-by: Vinish K B <Vinish_KorasseryBalakrishnan@comcast.com>

VNagori committed:
VPLAY-9175 - Create StreamAbstraction object in TuneHelper if it doesn't exist

Reason for change: Make the code more robust
Test Procedure: Run L1
Risks: Low

Change-Id: I6f4df4c958dfcaf5108666dd48b71f6f5b2d82ed
Signed-off-by: Vijendra Nagori <vinagori@synamedia.com>

DomSyna committed:
VPLAY-9393 HLS bitrate remains low post stream freeze

Reason for change: When the network bitrate drops low enough that
segments are dropped, when it recovers the bitrate doesn't always ramp
back up to the correct level.

There are 2 issues causing this:
1) The last profile index used could erroneously be restored, leaving
   AAMP playing the low bitrate playlist, but internally thinking it is
   playing a higher bitrate.
2) When segments are dropped the buffer depth calculation will be out by
   number of segments dropped * segment duration. This can then result
   in a -ve buffer depth calculation resulting in staying at a lower
   bitrate than can be supported.

Test procedure: Play HLS content and lower the NW bandwidth such that
the stream freezes, on restoring the NW bandwidth ensure AAMP ramps up
to the correct video quality. (I verified this from logs)

Risk: Low

Signed-off-by: Dominic Holden<doholden@synamedia.com>

Change-Id: Id4f2cb56b8f46963e3f9d251f00544744d888bb4

nebutler963 committed:
VPLAY-8897 - Rewind to start of TSB drops back to live edge on LLD channel

Reason for Change: When watching LLD content that has been watched for more than 25mins, on rewinding back to the beginning of the content, playback would start from the live edge.
Test Procedure: Run full stack, L1 and L2 test suites.

Risks: low
Change-Id: I8ec01482782204c6a9b987c658a326c985d6eea5

Deepikasri1234 committed:
RDK-56217 Decouple DRM License dependencies in AAMP

Reason for change: Fixed Godspell errors.
Test Procedure: Refer Ticket
Risks: Low
Priority: P1

Change-Id: I46e780bde7ff057c263c5a98147158e1fa4b5760
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Commit:
VPLAY-9412 Fix aamp build failure on Ubuntu

Reason for change: To add the missing #include <cstdint> header in middleware/PlayerLogManager.h resulting in AAMP build failure on Ubuntu
Risks: Low
Priority: P0

DEPS_BRANCH=rdk/components/generic/gst-plugins-rdk-aamp:dev_sprint_25_1
DEPS_BRANCH=rdk/components/generic/aampabr:dev_sprint_25_1
DEPS_BRANCH=rdk/components/generic/aampmetrics:dev_sprint_25_1

Change-Id: I5bbb4e2b531db6137c752ddaef8fb48014c96006
Signed-off-by: Rajat <Rajat_Yadav2@comcast.com>

Commit:
VPLAY-9183: UVE documentation clarification: showDiagnosticOverlay

Reason for change: Added sample code for showDiagnosticOverlay
Added changelog for 7.05
Test Procedure: None
Risks: None
Priority: P2

Change-Id: I136d0d9c3ebb55f77d50226b8f279dd5a4515610
Signed-off-by: Vinish K B <Vinish_KorasseryBalakrishnan@comcast.com>

pstroffolino committed:
VPLAY-9183 UVE documentation clarification: showDiagnosticOverlay

Reason for change : spelling fix and whitelist additions
Risks: Low
Test Procedure: try sample code in test app

Change-Id: I5caa67004694622bd121058ef16ae68743200e03

anjali-syna committed:
VPLAY-9407  AAMP TSB-Trickmode speed for higher rates are lower than expected

Reason for change: mUpcomingFragmentPosition is not properly updated for
trickmodes, which is resulting in not skipping the unwanted segments and
slowing down trick play, especially in higher rates

Changes:
* mUpcomingFragmentPosition is properly updated in AampTsbReader while
reading segments for trick play, so that unwanted segments can be
skipped and only the needed segments can be fed to gstreamer pipeline.

Test Procedure: Mentioned in the ticket

Risks: Low

Change-Id: Ieebbf0f654665f9dfddf1a4b37267ff321831027
Signed-off-by: Anjali Jacob <anjjacob@synamedia.com>

Deepikasri1234 committed:
VPLAY-9411 compilation warnings - 2505 sprint

Reason for change: address warnings surfacesd with xcode analysis and in l1test build logs
Test Procedure: confirm warnings no longer present
Priority: P1

Change-Id: I39bad7d163a3d87060906863a831ab33e4af7896
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

jfagunde committed:
VPLAY-9435  SEEKTOLIVE when FF to live for SLD linear channel

Reason for change: Send enteringLive event when FF to live for SLD
Summary of changes:
* Use the chunk cache for all content type if AAMP TSB is enabled
* SEEKTOLIVE when reaching EOS with AAMP TSB and no chunk mode
* Add SLD fragment to chunk cache when AAMP TSB enabled
* For standard mode, set CachedFragmentChunks size to MaxFragmentCached
* CacheFragment() logic uses chunk flag instead of low-latency
* When initializing the TSB reader, set the position to live play pos
* Create helper function to return live playing position
* In function Stop() clear AAMP TSB flags after stream teardown
* Add some new log lines
* Verify changes in L2 tests 5001 and 5002 and in L1 tests

Test Procedure: See Jira for how to check enteringLive event is sent

Risks: Low

Change-Id: I549b1274e40193b6d3b93208c07ce32cbc989b65
Signed-off-by: Jose Fagundez <jfagunde@synamedia.com>

pstroffolino committed:
VPLAY-9411 compilation warnings - 2505 sprint

Reason for change: godspell spelling fix
Test Procedure: no 'initiatecontent' spelling warning
Priority: P1

Change-Id: I435df04de3707fa8fd6029f372d71583731bdd14

Commit:
VPLAY-9373 : Ability to configure + / - av sync threshold values for MonitorAV.

Reason for change: added negative and positive moitor av threshold config
Test Procedure: check mointorav logs are coming or not
Priority: P1

Change-Id: Ie89f4c91698e2babba354bb88064bc53cb01d387
Signed-off-by: Gnanesha <Gnanesha_Thimmegowda@comcast.com>

Nithyak10 committed:
VPLAY-9421: Rate correction not working on LLD channels

Reason for change: Deprecating LiveLatencyCorrection() check as it is no
more used.
Test Procedure: Test steps in ticket
Risks: Low
Priority: P2

Change-Id: I7360503d3a63763cf9795be18513ef9250063ec3
Signed-off-by: Nithya Kasi <nithya_kasi@comcast.com>

LashminthaJ committed:
VPLAY-9079:remove remaining use of pthread

Reason for change:Removed the usage of pthread in aamp
Risks: Low
Test Procedure: Refer jira ticket
Priority: P1

Change-Id: Idbbfc7b401040141dc4d238ce21c04ecf0c08009
Signed-off-by: lashminthajayanathan <lashmintha_jayanathan@comcast.com>

LashminthaJ committed:
VPLAY-9185:Incorrect position calculated for Ad Reservation Events

Reason for change:Absolute reservation position of the ad is getting changed during ad completion
Risks: Low
Test Procedure: Refer jira ticket
Priority: P1

Change-Id: I860500d99b5183438ad669d50951946b3b11ec85
Signed-off-by: lashminthajayanathan <lashmintha_jayanathan@comcast.com>

Nithyak10 committed:
VPLAY-9193 : Middleware logs missing

Reason for change: SetLoggerInfo() API was called in wrong place,
possible fix added.
Test Procedure: Test steps in ticket
Risks: Low
Priority: P2

Change-Id: I9c10c1f5aa8db8c31b479c20d6f465184524edba
Signed-off-by: Nithya Kasi <nithya_kasi@comcast.com>

DomSyna committed:
VPLAY-9520  With AAMP TSB network issues cause return to home screen

Reason for change: With local AAMP TSB enabled if there is underflow the
AV will freeze regardless of if the NW recovers.

This is due to the logic that was added to improve pause, as the
segments are discarded after adding them to TSB in the belief that
playback will resume from TSB after a pause. When there is underflow,
the pipeline is also paused and this scenario was missed.

Test procedure: With Local TSB enabled tune to BBC1 and allow the device
to reach steady state. Lower the NW bandwidth to a point the requests
time out (10KB/s worked nicely for me). Once the AV freezes restore the
NW bandwidth and check AV resumes correctly.

Regression Test: Perform the same test as above but without the Local
TSB enabled and ensure the behaviour is unchanged.

Risk: LOW

Signed-off-by: Dominic Holden<doholden@synamedia.com>

Change-Id: I32c60946d4be89b9279629627c50f69af3b760bf

Commit:
RDKAAMP-3997  C interface for AAMP (required for KMP integration)

Reason for change: To create aamp-cli on Kotlin app
Risks: Low
Priority: P1

DEPS_BRANCH=rdk/components/generic/gst-plugins-rdk-aamp:dev_sprint_25_1
DEPS_BRANCH=rdk/components/generic/aampabr:dev_sprint_25_1
DEPS_BRANCH=rdk/components/generic/aampmetrics:dev_sprint_25_1

Change-Id: If553a60a3c2c933b24fefe260f6b4a60a95d3573
Signed-off-by: Rajat <Rajat_Yadav2@comcast.com>

Vinish100 committed:
VPLAY-8590: Implement AAMPEvent for reporting AV status

Reason for change: Define MonitorAVStatus event. Add
timer for periodically fetching AV status and implement
the JS listeners for same.
Test Procedure: As mentioned in the ticket
Risks: Low

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

pstroffolino committed:
VPLAY-9474 osx simulator - playback stuck on first frame when not using OpenGL shader for rendering

Reason for change: eliminate dependency on opengl/glut when building for OSX
Test Procedure: compile with and without USE_OPENGL
Risks: Low - simulator only

Change-Id: I39a372d6b6af5460e14c60f8e71cd0331c665eff

Commit:
Typos fixes

Change-Id: Ia335ded5023c4e4e49191d64af990c49fa5da947
Signed-off-by: Rajat <Rajat_Yadav2@comcast.com>

pstroffolino committed:
Revert "Typos fixes"

This reverts commit c1746ffe2622c84d6a397ae315e3bae63d4a0741.

Reason for revert: lacked correct commit message - reverting to re-add

Change-Id: I192609bf830848c8361e25c161fbae10650c9ddb

pstroffolino committed:
VPLAY-9536 C interface for AAMP (required for KMP integration

Reason for change: fix Godspell warnings about suspect spellings
Test Procedure: no Godspell typos cited
Risks: Low

Change-Id: I3f8bb5d7cf79247040901b90fc707068f6307d90

psiva01 committed:
VPLAY-9525: Switching between tune locators not working without explicit stop

Reason for change: Setting player state to IDLE at the start of Tune()
prevents internal Stop() for back to back tunes w/o explicit Stop. This was done to fix playback in reused instance after Stop. Moved this logic under RELEASED state check (if explicit Stop() already called), otherwise internal stop() is called so that back to back tunes work fine.
Test Procedure: See ticket
Risks: Medium

Change-Id: Ia6a75a6d78f822e98dea9b9b217d3e626599868e
Signed-off-by: sivasubramanian <sivasubramanian.patchaiperumal@ltts.com>

Commit:
VPLAY-9546 Fix -k option not working for install-aamp script

Reason for change: To fix -k option not working for install-aamp.sh script
Risks: Low
Priority: P0

DEPS_BRANCH=rdk/components/generic/gst-plugins-rdk-aamp:dev_sprint_25_1
DEPS_BRANCH=rdk/components/generic/aampabr:dev_sprint_25_1
DEPS_BRANCH=rdk/components/generic/aampmetrics:dev_sprint_25_1

Change-Id: Icf8ebf61adcd0d4f2ea853f80a21bc5e1584bc5e
Signed-off-by: Rajat <Rajat_Yadav2@comcast.com>

pstroffolino committed:
VPLAY-9557 deprecate NO_PLAYBIN compile-time option

Reason for change: tech debt - remove unused code
Test Procedure: regression only
Risks: Low

Change-Id: I3edf058f88831b9aa2155a1b4c44cf49ca7df3fd

pstroffolino committed:
VPLAY-9558 move gstMapDecoderLookUptable from InterfacePlayerRDK.h to InterfacePlayerRDK.cpp

Reason for change: tech debt: migrate from public header to to implementation
Test Procedure: regression only
Risks: Low

Change-Id: I2442d360e18eb667a0f55c2fc4e2a875a55dedd3

pstroffolino committed:
VPLAY-9560 SocInferface simplifications

Reason for change: tech debt: soc interface and identification simplifications
Test Procedure: regression only
Risks: Low

Change-Id: Ic12b8803273e90285d061e6efde77336b1e86b94

molakalapalliharipriya committed:
VPLAY-9065 : "set 5 0" command to silence volume

Reason for change: "set 5 0" command to silence volume is not working if included in aampcli.bat (but works if typed in aampcli post tune)
Test Procedure: Run aamp-cli with commands included in the batch file
Risks: Medium

Change-Id: I2bdcb7b40b8e85ab2f93ad55b6a30e13650d6a42
Signed-off-by: haripriya_molakalapalli <haripriya_molakalapalli@comcast.com>

pstroffolino committed:
VPLAY-9162 gst test harness support for GST_SEEK_FLAG_SEGMENT and non-flushing seek dai3() test, supporting per-period mid fragment clipping

Reason for change: flush fix needed for dai1 test
Test Procedure: dai1 test working without crash
Risks: Low

Change-Id: I5090ac3fdede9907441ac32de9ea66e030c56117

pstroffolino committed:
Revert "VPLAY-9079:remove remaining use of pthread"

This reverts commit 21505fe82abd9a5f7675a7deea4044aec7164569.

Reason for revert: build failure

Change-Id: Ia6894df4ac25d668bd9be452bf3759153efd3a69

pstroffolino committed:
Revert "Revert "VPLAY-9079:remove remaining use of pthread""

This reverts commit b8a435d1089049d52baa5100ac889b1c9306311b.

Reason for revert: reintroducing w/ fix

Change-Id: I4220f6f69f3fc4c89203ede02f9ff4cb773b77a9

DomSyna committed:
VPLAY-9566 [ENT-4082] Audio track change after FF to live plays from live edge instead of live playing position

Reason for change: When changing the prefered language whilst injecting
from the local TSB the wrong seek is being used. Resulting in seeking to
the very live edge, and not to the live position.

Test procedure: On a SLD channel with multiple languages (i.e 172) with fog dissabled, and local TSB
enabled:
 Pause for a couple of minutes
 FFWD to live edge so the automatic transition to play occurs
 Change the language from Welsh to English
 Observe play continues, there will be a small AV glitch as we are
effectively seeking to the same position as we are currently playing

Risk: Low

Signed-off-by: Dominic Holden<doholden@synamedia.com>

Change-Id: Id42d7c9bd9a98f913180e7ba77131e3df7762be1

Commit:
VPLAY-9553  Install openjdk-21-jre-headless in install-aamp script

Reason for change: To automate the installation of Java required for Kotlin build
Risks: Low
Priority: P2

DEPS_BRANCH=rdk/components/generic/gst-plugins-rdk-aamp:dev_sprint_25_1
DEPS_BRANCH=rdk/components/generic/aampabr:dev_sprint_25_1
DEPS_BRANCH=rdk/components/generic/aampmetrics:dev_sprint_25_1

Change-Id: I3385e9901b6353a711b57c01bdac0b2a5e2883df
Signed-off-by: Rajat <Rajat_Yadav2@comcast.com>

Vinish100 committed:
VPLAY-9299:Address 200ms tune delay when using enableMediaProcessor as true.

Reason for change: The delayed flush in media processor adds to tune delay.
Since SegmentTimeline based streams provide the PTS values in the manifest,
there is no need to parse the firstPTS from the segment.
Risks: Low
Test Procedure: Refer jira ticket
Priority: P1

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

shripadgithub committed:
RDKEMW-4539 : adding missed utest files

pstroffolino committed:
VPLAY-9512 add aamp configuration allowing new mp4demux to be used as alternative to qtdemux

Reason for change: enable A/B testing of qtdemux and new mp4demux
Test Procedure: see ticket
Risks: Low (off by default)

Change-Id: Idb6e17e2153146becb78042a5a57036862f5221d

pstroffolino committed:
VPLAY-9474 "osx simulator - playback stuck on first frame when not using OpenGL shader for rendering"

Reason for change: removed a block if #if 0 test code missed with checkin
Test Procedure: n/s - no functional change
Risks: None

Change-Id: I484b6018f6603415fbc8c8f1eeb56c5964605bc6

pstroffolino committed:
Revert "VPLAY-8897 - Rewind to start of TSB drops back to live edge on LLD channel"

This reverts commit 00e711ee3a9feafd7002a2d14f03fd45dd01bcb4.

Reason for revert: failing TEST_F(PrivAampTests, NotifyEOSReachedFFSeekPositionCalculation)

Change-Id: I98923d471b4690a4cdecb6fea2548c243601e1a4

pstroffolino committed:
VPLAY-9615 L2 thumbnail test 4004 failing

Reason for change: fix l2 test failure
Test Procedure: l2 test 4004 passing; no regression with (Peaocock) thumbnails
Risks: Low

Change-Id: I06d344f5605a0ad438fa8e4d0330a8c0f76e5a85

jfagunde committed:
VPLAY-9520  Add L1 test to verify delivered fix

Reason for change: Protect against regressions
Summary of changes:
* Extend L1 CacheFragment tests to include underflow
* Clarify what each run in L1 CacheFragment test is testing

Test procedure: Verify that L1 test MediaStreamContextTest pass

Risk: Low

Change-Id: I8ccb5dd5bc523b779691c36917175e7ee09d339b
Signed-off-by: Jose Fagundez <jfagunde@synamedia.com>

pstroffolino committed:
VPLAY-9435 AAMP is not sending enteringLive event for SLD content when fast-forwarding to live

Reason for change: avoid crash seen on OSX with deleted uninitialized pointer
Test Procedure: run l1 tests on OSX
Risks: Low

Change-Id: I893dcefccf3eba053f14d82f192cedb4c11c53e5

shripadgithub committed:
VPLAY-9624 Migration build error in dev_sprint_25_2

Reason for change: added gst-utils.cpp gst-utils.h in cmake found in diff with dev_sprint_25_1
Test Procedure:
Priority: P2
Risks:Low
Signed-off-by: shripad bankar <Shripad_Bankar@cable.comcast.com>

Abhi-jith-S committed:
Migration build error in dev_sprint_25_2

Signed-off-by: asasik397 <Abhijith_Sasikumar@comcast.com>

shripadbpersonal committed:
Merge pull request #35 from rdkcentral/hotfix/VPLAY-9624

VPLAY-9624 Migration build error in dev_sprint_25_2

anjali-syna committed:
VPLAY-9486 [ENT-4082] AD events missing when played back from TSB

Reason for change: Inaccuracy introduced into the ad event positions due to multiple casting from seconds (float) to milliseconds (uint64_t).

Changes:
* Changed storing position in milliseconds (as uint64_t) to use AampTime instead, to avoid casting

Test Procedure:

* Run L1 & L2 tests
* Run L3 test 2015 which demonstrated issue

Risks: Low

Change-Id: I382665e0c591f45078f827a0b43c04eea35bb092
Signed-off-by: Andrew Shepherd<anshephe@synamedia.com>

nebutler963 committed:
VPLAY-8897 - Rewind to start of TSB drops back to live edge on LLD channel

Reason for Change: When watching LLD content that has been watched for more than 25mins, on rewinding back to the beginning of the content, playback would start from the live edge.
Test Procedure: Run full stack, add L1 test.

Risks: low
Change-Id: Ide78b5c00eda90150ba4c6e1aaa01bcf1e6f13de

shripadgithub committed:
VPLAY-9624 : gsttestharness clean and removing unwanted words

Reason for change:
Test Procedure:
Priority: P2
Risks:Low
Signed-off-by: shripad bankar <Shripad_Bankar@cable.comcast.com>

Commit:
VPLAY-9883 : WPEWebProcess crash with PrivateInstanceAAMP::ProcessPendingDiscontinuity

Reason for Change: Do not delete stream abstraction instance
if pending discontinuity process is in progress
Test Procedure: Please see the ticket
Risks: Low
priority:P0

Change-Id: I11a23f2d483ed3fcaa3b3b517ed7264f93e78f46
Signed-off-by: Gnanesha <Gnanesha_Thimmegowda@comcast.com>

Vinish100 committed:
VPLAY-9643: Deprecate un-used config and compile-time macros

Reason for change: Deprecate un-used config and dead code in AAMP
Test Procedure: General sanity
Risks: Low

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

shripadbpersonal committed:
Merge pull request #43 from rdkcentral/feature/VPLAY-9512

VPLAY-9512 add aamp configuration allowing new mp4demux to be used as…

paulsm2-syna committed:
VPLAY-9681 - Correct logging line so AAMP test 2004 passes

Reason for change:
Correct logging line replace "\n" with a space

Test Procedure: None - tests are automated
Risks: Low

Vinish100 committed:
Merge branch 'dev_sprint_25_2' into vplay-9643

Vinish100 committed:
VPLAY-9643: Deprecate un-used config and compile-time macros

Reason for change: Remove un-used MACROs
Test Procedure: General sanity
Risks: Low

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

shripadbpersonal committed:
Revert "VPLAY-9512 add aamp configuration allowing new mp4demux to be used as…"

shripadbpersonal committed:
Merge pull request #48 from rdkcentral/revert-43-feature/VPLAY-9512

Revert "VPLAY-9512 add aamp configuration allowing new mp4demux to be used as…"

Vinish100 committed:
VPLAY-9643: Deprecate un-used config and compile-time macros

Reason for change: Remove un-used functions
Test Procedure: Sanity runs
Risks: None

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

Vinish100 committed:
Merge branch 'dev_sprint_25_2' into vplay-9643

paulsm2-syna committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9681

jfagunde committed:
VPLAY-9741 [ENT-4082] JSPP crashed while doing channel zapping

Reason for change: Fix a crash when doing channel change
Summary of changes:
* Clear the AAMP TSB flags before stopping and destroying the
  StreamAbstraction object.
* Acquire the Stream lock before stopping the StreamAbstraction object
* Use MediaTrack::abortInject flag for both cache (regular and chunk)
* Add aditional log lines when thread MonitorBufferHealth is created
  and destroyed

Test Procedure: Repeated channel changes and confirm there is no crash

Risks: Medium

Signed-off-by: Jose Fagundez <jfagunde@synamedia.com>

Commit:
Modified files for branch

anjali-syna committed:
VPLAY-9680[ENT-4082]AAMP TSB-Incorrect TSB duration in Progress Report

Reason for change: TSB duration in Progress Report is incorrect for few seconds
at the beginning of playback.

Changes:
* Properly updated TSB duration at the beginning of playback.

Test Procedure: Run L2 tests 5001 and 5002.

Risks: Low

pstroffolino committed:
Merge pull request #47 from rdkcentral/vplay-9643

VPLAY-9643: Deprecate un-used config and compile-time macros

Vinish100 committed:
VPLAY-9299:Address 200ms tune delay when using enableMediaProcessor as true.

Reason for change: Add support for a passThroughMode in isobmff processor
when segment timeline is enabled. In PT mode, AAMP will only parse the PTS
and timeScale for certain usecases and withhold caching and flush
Risks: Low
Test Procedure: Refer jira ticket
Priority: P1

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

TanujSharma369 committed:
VPLAY-9732: Fix crash on retuning to AES-encrypted channel

Commit:
VPLAY-9677 [ENT-4082] AAMP TSB - Configuration Enabled/Disabled

Reason For Change: AAMP TSB can be enabled 2 ways in the config

Test Procedure: L1 and L2 5004 and 5002

Change-Id: Ic950b5cf4c52dc1676d331fac7c3c6079c392936

pstroffolino committed:
Merge pull request #58 from rdkcentral/feature/VPLAY-9732

VPLAY-9732: Fix crash on retuning to AES-encrypted channel

tabbas651 committed:
VPLAY-9716: Removal of WPEFrameworkSecurity Agent Utility (#56)

Reason for change: Added DISABLE_SECURITY_TOKEN Flag to disable the WPEFrameworkSecurity Token generation changes
Test Procedure: please referred from the ticket
Risks: Medium

Signed-off-by: Thamim Razith Abbas Ali <ThamimRazith_AbbasAli@comcast.com>
Co-authored-by: shripadbpersonal <105402691+shripadbpersonal@users.noreply.github.com>

pstroffolino committed:
VPLAY-9708 aamp not generating ala carte selectable schemes for individual l1 tests

Reason for change: allow utests to again be selectable in xcode
Test Procedure: select and run individual utests from xcode
Risks: Low

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

Vinish100 committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9299

rajat698 committed:
Merge pull request #61 from rdkcentral/feature/VPLAY-9708-xcode-utests

VPLAY-9708 aamp not generating ala carte selectable schemes for indiv…

pstroffolino committed:
VPLAY-9710 aampcli kmp build failing to compile for older x86 Intel Mac

Reason for Change: use gst_macos_main_simple workaround only on newer macs
Test Procedure: compile on x86 mac
Risks: Low

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

rajat698 committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9710-kmpx86

rajat698 committed:
Merge pull request #63 from rdkcentral/feature/VPLAY-9710-kmpx86

VPLAY-9710 aampcli kmp build failing to compile for older x86 Intel Mac

anjali-syna committed:
VPLAY-9680 [ENT-4082] AAMP TSB - TSB duration in Progress Report is incorrect

Reason for change: TSB duration in Progress Report is incorrect for
few seconds at the beginning of playback.

Changes:
* Properly updated TSB duration at the beginning of playback.

Test Procedure: Run L1 tests and L2 tests - 5001 and 5002.

Risks: Low

Signed-off-by: anjali-syna <206662904+anjali-syna@users.noreply.github.com>

anjali-syna committed:
Incorporating review comments

Signed-off-by: anjali-syna <206662904+anjali-syna@users.noreply.github.com>

anjali-syna committed:
Addressing review comments

Signed-off-by: anjali-syna <206662904+anjali-syna@users.noreply.github.com>

Vinish100 committed:
VPLAY-9299: Address 200ms tune delay when using enableMediaProcessor as true

Reason for change: Add L1 tests for PTM mode in media processor
Test Procedure: Validate all L1 tests pass
Risks: None

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

nebutler963 committed:
VPLAY-9391 -AAMP TSB - OOB subtitles

Reason for Change: OOB subtitles support when using TSB.
Test Procedure: L1 & L2.  Full stack validation.
Risks: Low

Vinish100 committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9299

pstroffolino committed:
Merge pull request #55 from rdkcentral/feature/VPLAY-9299

VPLAY-9299:Address 200ms tune delay when using enableMediaProcessor as true

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9680

pstroffolino committed:
Merge pull request #65 from rdkcentral/feature/VPLAY-9680

vplay 9680 - Incorrect TSB duration in Progress Report

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9391

pstroffolino committed:
Merge pull request #71 from rdkcentral/feature/VPLAY-9391

VPLAY-9391 -AAMP TSB - OOB subtitles

pstroffolino committed:
VPLAY-9747 spelling fixes

Reason for change: fix godspell reported issues
Test Procedure: check godspell results
Risks: Low

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9566

shripadbpersonal committed:
Merge pull request #74 from rdkcentral/feature/VPLAY-9747-spelling

VPLAY-9747 spelling fixes

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9566

pstroffolino committed:
Merge pull request #69 from rdkcentral/feature/VPLAY-9566

VPLAY-9566 [ENT-4082] Audio track change after FF to live plays from …

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9741

paulsm2-syna committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9681

pstroffolino committed:
Merge pull request #67 from rdkcentral/feature/VPLAY-9741

VPLAY-9741 [ENT-4082] JSPP crashed while doing channel zapping

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9486

pstroffolino committed:
Merge pull request #70 from rdkcentral/feature/VPLAY-9486

VPLAY-9486 - AD events missing when played back from TSB

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9677

psiva01 committed:
VPLAY-9678: Crash(60223193) observed during channel change on linear channels

Reason for change: Added ERROR or IDLE state check guard before calling
GetAvailableAudioTracks() to avoid this crash. Added L1 to catch this.
Test Procedure: Refer Jira
Risks: Low

Signed-off-by: Sivasubramanian <sivasubramanian.patchaiperumal@ltts.com>

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9681

pstroffolino committed:
Merge pull request #76 from rdkcentral/feature/VPLAY-9677

VPLAY-9677 [ENT-4082] AAMP TSB - Configuration Enabled/Disabled

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9681

pstroffolino committed:
Merge pull request #44 from rdkcentral/feature/VPLAY-9681

VPLAY-9681 - Correct logging line so AAMP test 2004 passes

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9678

Commit:
VPLAY-9758 L1 test integration in Github

Reason for change: To add L1 testing to Github CI
Risks: Low
Priority: P1

Signed-off-by: Rajat <Rajat_Yadav2@comcast.com>

Commit:
VPLAY-9758 L1 test integration in Github

Reason for change: To add L1 testing to Github CI
Risks: Low
Priority: P1

Signed-off-by: Rajat <Rajat_Yadav2@comcast.com>

jfagunde committed:
VPLAY-9391 [ENT-4082] Allow text track change with AAMP TSB enabled

Reason for change: Allow text track change with AAMP TSB enabled

Test Procedure: Enable AAMP TSB and switch text track

Risks: Low

Signed-off-by: Jose Fagundez <jfagunde@synamedia.com>

Vinish100 committed:
VPLAY-8590: Implement AAMPEvent for reporting AV status

Reason for change: Update UVE documentation for MonitorAV events
Test Procedure: As mentioned in the ticket
Risks: None

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

shripadbpersonal committed:
Merge pull request #99 from rdkcentral/dev_sprint_25_2

Dev sprint 25 2 rebase

pstroffolino committed:
Merge pull request #97 from rdkcentral/feature/VPLAY-9391

VPLAY-9391 [ENT-4082] Allow text track change with AAMP TSB enabled

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9678

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-8590

pstroffolino committed:
Merge pull request #96 from rdkcentral/feature/VPLAY-8590

VPLAY-8590: Implement AAMPEvent for reporting AV status

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9678

pstroffolino committed:
Merge pull request #82 from rdkcentral/feature/VPLAY-9678

VPLAY-9678: Crash(60223193) observed during channel change on linear …

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9758

pstroffolino committed:
Merge pull request #85 from rdkcentral/feature/VPLAY-9758

Feature/vplay 9758

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9410

pstroffolino committed:
Merge pull request #49 from rdkcentral/feature/VPLAY-9410

VPLAY-9410:muxed hls ts playback issues after pts rollover

Commit:
RDK-57972 : 2506 Federated Merge 2.6.1

Reason for change: 2506 Federated Merge 2.6.1
Test Procedure: Build and Verify
Risks: Low
Priority: P2

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rdkcentral/aamp?pullRequest=104) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/rdkcentral/aamp?pullRequest=104) before we can accept your contribution.<br/>**12** out of **14** committers have signed the CLA.<br/><br/>:white_check_mark: jfagunde<br/>:white_check_mark: rajat698<br/>:white_check_mark: anjali-syna<br/>:white_check_mark: Vinish100<br/>:white_check_mark: nebutler963<br/>:white_check_mark: p-bond<br/>:white_check_mark: DomSyna<br/>:white_check_mark: pstroffolino<br/>:white_check_mark: psiva01<br/>:white_check_mark: paulsm2-syna<br/>:white_check_mark: paulpandian25<br/>:white_check_mark: shripadbpersonal<br/>:x: Rajat<br/>:x: Deepikasri1234<br/><hr/>**Rajat** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rdkcentral/aamp?pullRequest=104) it.</sub>

Vinish100 committed:
Revert "VPLAY-9299: Address 200ms tune delay when using enableMediaProcessor as true"

This reverts commit edac63eb4c7ddfafef368ffceb1c723312e5f345.
This reverts commit 056e0e50561b845dd1bbcc34967a59280218b061.

Revert "VPLAY-9299:Address 200ms tune delay when using enableMediaProcessor as true."

This reverts commit dd892c594eb8bd6adb2eb5cc43680d6c7a86b505.

Commit:
VPLAY-9799 : Negative Buffer value seen on performing Go to live/FFWD after rewind operation on linear channels / SLE contents
Revert "VPLAY-9393 HLS bitrate remains low post stream freeze"

Reason for Change: Revert "VPLAY-9393 HLS bitrate remains low post stream freeze" which is causing the issue
Test Procedure: Please see the ticket
Risks: Low
priority:P1
This reverts commit 633f0b67c48230438cfe3764bf8d490d379d138e.

psiva01 committed:
VPLAY-9627: Crash(29398566) observed during channel change on linear channels (#135)

Reason for change: Changed DrmSessionManager::createDrmSession() to check for
cachedKeyIDs before updating it's failed status during failure. This will avoid
crash due to race condition caused by stop()
Test Procedure: Refer Jira
Risks: Low
Priority: P1

Signed-off-by: Sivasubramanian <sivasubramanian.patchaiperumal@ltts.com>
Co-authored-by: pstroffolino <Philip_Stroffolino@cable.comcast.com>

paulpandian25 committed:
Merge pull request #138 from rdkcentral/feature/VPLAY-9299_Revert_2506

Revert "VPLAY-9299:Address 200ms tune delay when using enableMediaProcessor as true."

jfagunde committed:
VPLAY-9842 Not enable AAMP TSB for FOG urls

Reason for Change: Disable AAMP TSB if FOG is used
Test Procedure: Please see the ticket
Risks: Low

Signed-off-by: Jose Fagundez <jfagunde@synamedia.com>

Add manifest parameter to IsFogUrl()

Add L1 tests

shripadbpersonal committed:
Merge pull request #181 from rdkcentral/feature/RDKEMW-5467

RDKEMW-5467 : Set PlayerState to IDLE during stop

Deepikasri1234 committed:
VPLAY-9906 [Foxtel][AAMP]Crash is observed during 4K VOD playback in XiOne-Foxtel

Reason for change: Updating Curldownloader instances upon calling UpdateMaxDrmSession
Test Procedure: Refer Ticket
Risks: Low
Priority: P1

Change-Id: I2b4c22d9a381f792eb08dfe8a49eecabe6f374d6
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
VPLAY-9906 [Foxtel][AAMP]Crash is observed during 4K VOD playback in XiOne-Foxtel

Reason for change: Updating Curldownloader instances upon calling UpdateMaxDrmSession
Test Procedure: Refer Ticket
Risks: Low
Priority: P1

Change-Id: I2b4c22d9a381f792eb08dfe8a49eecabe6f374d6
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

VPLAY-9906 [Foxtel][AAMP]Crash is observed during 4K VOD playback in XiOne-Foxtel

Reason for change: Addressed review comments.
Test Procedure: Refer Ticket
Risks: Low

psiva01 committed:
Revert "VPLAY-9906 [Foxtel][AAMP]Crash is observed during 4K VOD playback in XiOne-Foxtel
This reverts commit 5efc21c7a1a74558f7a66090915d7903163421af.

paulpandian25 committed:
Merge branch 'develop' into feature/RDK-57972_federated_2506_develop
